seed/views/v3/import_files.py

```diff
@@ -0,0 +1,808 @@
+                    exclude=['extra_data']
+                )
+
+                prop_dict.update(
+                    TaxLotProperty.extra_data_to_dict_with_mapping(
+                        prop.extra_data,
+                        property_column_name_mapping,
+                        fields=prop.extra_data.keys(),
+                    ).items()
+                )
+                property_results.append(prop_dict)
+
+            result['properties'] = property_results
+
+        if inventory_type == 'taxlots' or inventory_type == 'all':
+            tax_lots = TaxLotState.objects.filter(
+                import_file_id=import_file_id,
+                data_state__in=[DATA_STATE_MAPPING, DATA_STATE_MATCHING],
+                merge_state__in=[MERGE_STATE_UNKNOWN, MERGE_STATE_NEW]
+            ).only(*fields['TaxLotState']).order_by('id')
+
+            tax_lot_results = []
+            for tax_lot in tax_lots:
+                tax_lot_dict = TaxLotProperty.model_to_dict_with_mapping(
+                    tax_lot,
+                    taxlot_column_name_mapping,
+                    fields=fields['TaxLotState'],
+                    exclude=['extra_data']
+                )
+                tax_lot_dict.update(
+                    TaxLotProperty.extra_data_to_dict_with_mapping(
+                        tax_lot.extra_data,
+                        taxlot_column_name_mapping,
+                        fields=tax_lot.extra_data.keys(),
+                    ).items()
+                )
+                tax_lot_results.append(tax_lot_dict)
+
+            result['tax_lots'] = tax_lot_results
+
+        return result
+
+    @staticmethod
+    def has_coparent(state_id, inventory_type, fields=None):
+        """
+        Return the coparent of the current state id based on the inventory type. If fields
+        are given (as a list), then it will only return the fields specified of the state model
+        object as a dictionary.
+
+        :param state_id: int, ID of PropertyState or TaxLotState
+        :param inventory_type: string, either properties | taxlots
+        :param fields: list, either None or list of fields to return
+        :return: dict or state object, If fields is not None then will return state_object
+        """
+        state_model = PropertyState if inventory_type == 'properties' else TaxLotState
+
+        # TODO: convert coparent to instance method, not class method
+        audit_entry, audit_count = state_model.coparent(state_id)
+
+        if audit_count == 0:
+            return False
+
+        if audit_count > 1:
+            return JsonResponse(
+                {
+                    'status': 'error',
+                    'message': 'Internal problem occurred, more than one merge record found'
+                },
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR
+            )
+
+        return audit_entry[0]
+
+    @swagger_auto_schema(
+        request_body=AutoSchemaHelper.schema_factory({
```

Comment:
Seeing it in _action_ - what do you think about renaming this to "body_schema_factory"? If you agree with the change, we could also just do it in the next PR, where the other v3 endpoints are built out.

Reply:
I might prefer to keep it generic, as there's nothing relating this directly to the body. In #2247 I also use it to create _response_ schemas as well (rather than a serializer).

Reply:
Ah ok, didn't realize it could be used in that way. Agree - let's keep it as is. 👍